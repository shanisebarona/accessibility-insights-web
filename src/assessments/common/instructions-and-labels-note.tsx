// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import * as Markup from '../markup';

export const InstructionsAndLabelsNotes = () => (
    <Markup.Emphasis>
        Note: Both WCAG 2.0 and 2.1 require a widget's visible label and instructions (if present)
        to be programmatically determinable. WCAG 2.1 also requires a widget's visible label and
        instructions (if present) to be included in its accessible name and description.
    </Markup.Emphasis>
);
