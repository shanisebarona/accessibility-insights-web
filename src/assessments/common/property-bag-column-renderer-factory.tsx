// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { propertyBagColumnRenderer } from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: InstanceTableRow<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
