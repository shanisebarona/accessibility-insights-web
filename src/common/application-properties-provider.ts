// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ToolData } from './types/store-data/unified-data-interface';

export const createToolData = (
    scanEngineName: string,
    scanEngineVersion: string,
    toolName: string,
    toolVersion?: string,
    environmentName?: string,
    resolution?: string,
): ToolData => {
    return {
        scanEngineProperties: {
            name: scanEngineName,
            version: scanEngineVersion,
        },
        applicationProperties: {
            name: toolName,
            version: toolVersion,
            environmentName: environmentName,
            resolution: resolution,
        },
    };
};
