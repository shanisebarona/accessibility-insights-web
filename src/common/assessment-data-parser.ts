// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';

export class AssessmentDataParser {
    public parseAssessmentData = (assessmentData: string): VersionedAssessmentData => {
        const loadedAssessmentData: VersionedAssessmentData = JSON.parse(assessmentData);
        return loadedAssessmentData;
    };
}
