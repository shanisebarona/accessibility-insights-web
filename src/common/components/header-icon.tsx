// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { BrandBlue } from '../../icons/brand/blue/brand-blue';
import { BrandWhite } from '../../icons/brand/white/brand-white';
import { NamedFC } from '../react/named-fc';
import { ThemeInnerState } from './theme';
import {
    withStoreSubscription,
    WithStoreSubscriptionDeps,
    WithStoreSubscriptionProps,
} from './with-store-subscription';

export type HeaderIconState = ThemeInnerState;

export type HeaderIconDeps = WithStoreSubscriptionDeps<HeaderIconState>;

export type HeaderIconProps = {
    invertColors?: boolean;
} & WithStoreSubscriptionProps<HeaderIconState>;

export const HeaderIconComponent = NamedFC<HeaderIconProps>('HeaderIconComponent', props => {
    const state = props.storeState.userConfigurationStoreData;
    const enableHighContrast = state && state.enableHighContrast;

    const { invertColors = false } = props;

    const defaultIcon = invertColors ? <BrandBlue /> : <BrandWhite />;
    const highContrastIcon = invertColors ? <BrandWhite /> : <BrandBlue />;

    return enableHighContrast ? highContrastIcon : defaultIcon;
});

export const HeaderIcon =
    withStoreSubscription<HeaderIconProps, HeaderIconState>(HeaderIconComponent);
