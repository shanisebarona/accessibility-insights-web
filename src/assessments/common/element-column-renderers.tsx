// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { AssessmentInstanceDetailsColumn } from 'DetailsView/components/assessment-instance-details-column';
import * as React from 'react';

export function onRenderPathColumn(item: InstanceTableRow): JSX.Element {
    let textContent = '';
    if (item.instance.target) {
        textContent = item.instance.target.join(';');
    }

    return (
        <AssessmentInstanceDetailsColumn
            textContent={textContent}
            customClassName="not-applicable"
        />
    );
}

export function onRenderSnippetColumn(item: InstanceTableRow): JSX.Element {
    return (
        <AssessmentInstanceDetailsColumn
            textContent={item.instance.html}
            customClassName="not-applicable"
        />
    );
}
