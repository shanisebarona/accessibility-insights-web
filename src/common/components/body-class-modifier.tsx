// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DocumentManipulator } from 'common/document-manipulator';
import * as React from 'react';

export type BodyClassModifierDeps = {
    documentManipulator: DocumentManipulator;
};

export type BodyClassModifierProps = {
    classNames: string[];
    deps: BodyClassModifierDeps;
};

// We use our own BodyClassModifier rather than using Helmet's similar functionality because we
// need multiple BodyClassModifiers to be able to co-exist without overwriting each other or the
// body classes office fabric maintains.
export class BodyClassModifier extends React.Component<BodyClassModifierProps> {
    public render(): JSX.Element | null {
        return null;
    }

    public componentDidMount(): void {
        this.updateClassNames([], this.props.classNames);
    }

    public componentDidUpdate(prevProps: BodyClassModifierProps): void {
        this.updateClassNames(prevProps.classNames, this.props.classNames);
    }

    public componentWillUnmount(): void {
        this.updateClassNames(this.props.classNames, []);
    }

    private updateClassNames(oldClassNames: string[], newClassNames: string[]): void {
        const { documentManipulator } = this.props.deps;

        const addedClassNames = newClassNames.filter(newName => !oldClassNames.includes(newName));
        const removedClassNames = oldClassNames.filter(oldName => !newClassNames.includes(oldName));

        // document.body may include classes not managed by this BodyClassModifier; leave them as-is
        let allBodyClassNames = documentManipulator.getBodyClassNames();
        allBodyClassNames = allBodyClassNames.filter(n => !removedClassNames.includes(n));
        allBodyClassNames.push(...addedClassNames);

        documentManipulator.setBodyClassNames(allBodyClassNames);
    }
}
