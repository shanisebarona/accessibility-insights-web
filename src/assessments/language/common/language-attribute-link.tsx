// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NewTabLink } from 'common/components/new-tab-link';
import * as React from 'react';

export const LanguageAttributeLink = () => (
    <NewTabLink href="http://www.iana.org/assignments/language-subtag-registry/language-subtag-registry">
        language attribute
    </NewTabLink>
);
