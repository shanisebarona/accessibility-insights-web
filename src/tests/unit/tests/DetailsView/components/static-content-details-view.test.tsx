// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import { IMock, It, Mock, Times } from 'typemoq';

import { ContentReference } from 'views/content/content-page';
import { VisualizationToggle } from '../../../../../common/components/visualization-toggle';
import {
    StaticContentDetailsView,
    StaticContentDetailsViewProps,
} from '../../../../../DetailsView/components/static-content-details-view';
import { BaseDataBuilder } from '../../../common/base-data-builder';
import { EventStubFactory, NativeEventStub } from '../../../common/event-stub-factory';

describe('StaticContentDetailsViewTest', () => {
    it('renders content page component', () => {
        const props = new StaticContentDetailsViewPropsBuilder().build();

        const actual = shallow(<StaticContentDetailsView {...props} />);

        expect(actual.debug()).toMatchSnapshot();
    });

    it('click the toggle', () => {
        const event = new EventStubFactory().createMouseClickEvent() as any;
        const clickHandlerMock = Mock.ofInstance(theEvent => {});
        clickHandlerMock.setup(chm => chm(event)).verifiable(Times.once());

        const propsBuilder = new StaticContentDetailsViewPropsBuilder().setupOnToggleClickMock(
            event,
        );
        const props: StaticContentDetailsViewProps = propsBuilder.build();
        const testObject = shallow(<StaticContentDetailsView {...props} />);
        const visualizationToggle = testObject.find(VisualizationToggle);
        visualizationToggle.prop('onClick')(event);
        propsBuilder.verifyAll();
    });
});

class StaticContentDetailsViewPropsBuilder extends BaseDataBuilder<StaticContentDetailsViewProps> {
    private onToggleClickMock: IMock<(event) => void> = Mock.ofInstance(event => {});

    constructor() {
        super();

        this.data = {
            title: 'my test title',
            visualizationEnabled: true,
            toggleLabel: 'my test toggle label',
            onToggleClick: this.onToggleClickMock.object,
            content: Mock.ofType<ContentReference>().object,
            guidance: Mock.ofType<ContentReference>().object,
            stepsText: 'test steps text',
        } as StaticContentDetailsViewProps;
    }

    public setupOnToggleClickMock(event: NativeEventStub): StaticContentDetailsViewPropsBuilder {
        this.onToggleClickMock.setup(click => click(It.isValue(event))).verifiable(Times.once());

        return this;
    }

    public verifyAll(): void {
        this.onToggleClickMock.verifyAll();
    }
}
