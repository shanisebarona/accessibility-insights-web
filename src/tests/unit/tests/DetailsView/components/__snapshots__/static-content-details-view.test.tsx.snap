// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticContentDetailsViewTest renders content page component 1`] = `
"<div className=\\"staticContentInDetailsView\\">
  <h1>
    my test title
     test steps text 
    <ContentLink deps={[undefined]} reference={[Function: function () {
                        var args = [];
                        for (var _i = 0; _i < arguments.length; _i++) {
                            args[_i] = arguments[_i];
                        }
                        _this._interceptor.removeInvocation(invocation_1);
                        var method = new MethodInfo(target, p);
                        var methodInvocation = new MethodInvocation(target, method, args, ProxyType.DYNAMIC);
                        _this._interceptor.intercept(methodInvocation);
                        return methodInvocation.returnValue;
                    }]} iconName=\\"info\\" />
  </h1>
  <VisualizationToggle checked={true} onClick={[Function: proxy]} label=\\"my test toggle label\\" className=\\"detailsViewToggle\\" visualizationName=\\"my test title\\" />
  <ContentInclude deps={[undefined]} content={[Function: function () {
                      var args = [];
                      for (var _i = 0; _i < arguments.length; _i++) {
                          args[_i] = arguments[_i];
                      }
                      _this._interceptor.removeInvocation(invocation_1);
                      var method = new MethodInfo(target, p);
                      var methodInvocation = new MethodInvocation(target, method, args, ProxyType.DYNAMIC);
                      _this._interceptor.intercept(methodInvocation);
                      return methodInvocation.returnValue;
                  }]} />
</div>"
`;
